app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java

```diff
@@ -393,17 +392,27 @@ public Task<HttpsCallableResult> findPin(String pid, Location location) {
         data.put("latitude", location.getLatitude());
         data.put("longitude", location.getLongitude());
 
-        return functions.getHttpsCallable("findPin").call(data).addOnSuccessListener(task -> {
-            // TODO: make cloud function return reward & update cached pinnie count with it
-            //noinspection unchecked
-            Map<String, String> result = (Map<String, String>) task.getData();
-            //noinspection ConstantConditions
-            String broadLocationName = result.get("broadLocationName");
-            String nearbyLocationName = result.get("nearbyLocationName");
-            activity.add(new ActivityItem(auth.getUid(), pid, ActivityItem.ActivityType.FIND,
-                    broadLocationName, nearbyLocationName));
-            foundPinMetadata.add(new PinMetadata(pid, broadLocationName, nearbyLocationName));
-        }).addOnFailureListener(e -> Log.w(TAG, "Error finding pin.", e));
+        return functions.getHttpsCallable("findPin")
+                .call(data)
+                .continueWith(task -> {
+                    //noinspection unchecked
+                    Map<String, Object> result = (Map<String, Object>) task.getResult().getData();
+
+                    //noinspection ConstantConditions
+                    String broadLocationName = (String) result.get("broadLocationName");
+                    String nearbyLocationName = (String) result.get("nearbyLocationName");
+                    Long reward = Long.parseLong(result.getOrDefault("reward", 0L).toString());
+                    Log.d(TAG, String.format("Got reward for pin: %d", reward));
+
+                    activity.add(new ActivityItem(auth.getUid(), pid, ActivityItem.ActivityType.FIND,
+                            broadLocationName, nearbyLocationName));
+                    foundPinMetadata.add(new PinMetadata(pid, broadLocationName, nearbyLocationName));
+
+                    return reward;
```

Comment:
what if we just update the cached pinnies in this `continueWith` clause so we don't need to create another public method that exposes  modifying the pinnie count?

Reply:
Good call

Reply:
Change has been made